### PR TITLE
chore(deps): constrain pyarrow via `snowflake-sqlalchemy` by adding the `pandas` extra

### PR DESCRIPTION
**Pull request by cpcloud**
This PR prevents an incompatible version of pyarrow being installed when using the snowflake backend. As a bonus, we no longer need to hack it pyarrow@8 in the Snowflake CI setup.

### CODE REVIEW

**Comment by cpcloud** [comment]
There's a PR to update `snowflake`'s python API to use arrow 10 (https://github.com/snowflakedb/snowflake-connector-python/pull/1336).

**Comment by cpcloud** [comment]
We'll pick this up automatically when they do the bump and cut a new release.

**Comment by cpcloud** [comment]
Without this constraint, segfaults abound if you just `pip install pyarrow` without a version constraint.

File: pyproject.toml [hunk 1]
```diff
@@ -65,9 +65,14 @@ pyspark = { version = ">=3,<4", optional = true }
 regex = { version = ">=2021.7.6", optional = true }
 requests = { version = ">=2,<3", optional = true }
 Shapely = { version = ">=1.6,<1.8|>=1.9,<2", optional = true }
+snowflake-connector-python = { version = ">=2.7.10,<3", optional = true, extras = [
+  "pandas"
+] }
+snowflake-sqlalchemy = { version = ">=1.4.1,<2", optional = true, extras = [
+  "pandas"
+] }
```
Comment (gforsyth) [inline pyproject.toml:73]: Can you add a comment about why we have to do this crazy business?

Comment (cpcloud) [reply pyproject.toml:73]: Which part? Adding the transitive dependency, or something else?

Comment (cpcloud) [reply pyproject.toml]: Done!

**Review by gforsyth — APPROVED** [review]

**Pull request merged**